packages/openstef-models/src/openstef_models/models/forecasting/base_case_forecaster.py

```diff
@@ -0,0 +1,305 @@
+                   If None, uses default configuration with 7-day primary and 14-day fallback lags.
+        """
+        self._config = config or BaseCaseForecasterConfig()
+        self._lag_columns_detected: bool = False
+        self._primary_lag_target_column_name: str | None = None
+        self._fallback_lag_target_column_name: str | None = None
+
+    @property
+    @override
+    def config(self) -> BaseCaseForecasterConfig:
+        return self._config
+
+    @property
+    @override
+    def hyperparams(self) -> BaseCaseForecasterHyperParams:
+        return self._config.hyperparams
+
+    @override
+    def to_state(self) -> State:
+        return {
+            "version": MODEL_CODE_VERSION,
+            "config": self.config.model_dump(mode="json"),
+        }
+
+    @override
+    def from_state(self, state: State) -> Self:
+        if not isinstance(state, dict) or "version" not in state or state["version"] > MODEL_CODE_VERSION:
+            raise ModelLoadingError("Invalid state for BaseCaseForecaster")
+
+        return self.__class__(config=BaseCaseForecasterConfig.model_validate(state["config"]))
+
+    @staticmethod
+    def _construct_lag_column_name(target_column: str, lag: timedelta) -> str:
+        """Constructs the column name for the lag target feature.
+
+        Args:
+            target_column: The name of the target column
+            lag: The timedelta lag
+
+        Returns:
+            The lag feature column name
+        """
+        # Convert timedelta to ISO format (e.g., P7D becomes -P7D for negative lag)
+        iso_lag = timedelta_to_isoformat(-lag)
+        return f"{target_column}_lag_{iso_lag}"
+
+    @property
+    @override
+    def is_fitted(self) -> bool:
+        return self._lag_columns_detected
+
+    def _detect_lag_columns(self, data: ForecastInputDataset) -> None:
+        """Detect lag columns in the input data.
+
+        Looks for columns matching patterns like 'load_lag_-P7D', 'load_lag_-P14D', etc.
+
+        Args:
+            data: The input dataset containing features and target variable.
+
+        Raises:
+            MissingColumnsError: If both primary and fallback lag columns are not available.
+        """
+        self._primary_lag_target_column_name = self._construct_lag_column_name(
+            data.target_column, self._config.hyperparams.primary_lag
+        )
+
+        self._fallback_lag_target_column_name = self._construct_lag_column_name(
+            data.target_column, self._config.hyperparams.fallback_lag
+        )
+
+        # Check if we have at least one lag column available
+        has_primary = self._primary_lag_target_column_name in data.feature_names
+        has_fallback = self._fallback_lag_target_column_name in data.feature_names
+
+        if not has_primary and not has_fallback:
+            raise MissingColumnsError([self._primary_lag_target_column_name, self._fallback_lag_target_column_name])
```

Comment:
Perhaps we could use https://github.com/OpenSTEF/openstef/blob/release/v4.0.0/packages/openstef-core/src/openstef_core/datasets/validation.py#L23 `validate_required_columns` here.